src/compute/__tests__/mean.test.ts

```diff
@@ -65,3 +66,27 @@ test('mean from points in rgba image', () => {
 
   expect(result).toStrictEqual([1, 2, 3, 0]);
 });
+test('must throw if array is empty', () => {
+  const image = testUtils.createRgbaImage([
+    [1, 2, 3, 0],
+    [1, 2, 3, 0],
+  ]);
+  const points: Point[] = [];
+
+  expect(() => {
+    const result = image.mean({ points });
+    return result;
+  }).toThrow('Array of coordinates is empty.');
+});
+test("must throw if point's coordinates are invalid.", () => {
+  const image = testUtils.createRgbaImage([
+    [1, 2, 3, 0],
+    [1, 2, 3, 0],
+  ]);
+  const points: Point[] = [{ column: 0, row: 2 }];
```

Comment:
But if you pass `{column: 4, row:0}` which is also out of bound it does not throw right?

Can you add a test case +  change the implementation so that it also throws in that case?

Reply:
> But if you pass `{column: 4, row:0}` which is also out of bound it does not throw right?
> 
> Can you add a test case + change the implementation so that it also throws in that case?

I think the way is to add a comparison between `point.column` and `image.width`. I am not sure I understand now what is the efficiency advantage of calculating the index in the first place. We can just use `point.column` and `point.row` to compare and then use either `getValue(point.column,point.row,channel)` or `getValueByPoint(point,channel)`.

Reply:
Yes there is no point anymore in calculating the index separately.